### PR TITLE
fix(*): fix security vulnerability and database selection

### PR DESCRIPTION
**Pull request by outsinre**
1. If two redis clients share the same connection but use different database, both clients might receive error upon set/get data.
2. If a client without authentication shares a connection that is authenticated by other clients, this client bypass ACL.

FTI-5839

### CODE REVIEW

**Comment by bungle** [comment]
@outsinre could you add tests?

**Comment by bungle** [comment]
@outsinre can you also explain security vulnerability? I guess this is somewhat related to connection pooling that can be shared by some other app?

**Comment by bungle** [comment]
This library supports `pool_name`:
https://github.com/bungle/lua-resty-session/blob/master/lib/resty/session/redis.lua#L191

So I guess it is just the default pool name generation?

File: lib/resty/session/utils.lua [hunk 1]
```diff
@@ -774,6 +775,56 @@ local hmac_sha256 do
...
+    end
+
+    local hex_mac
+    hex_mac, err = to_hex(digest)
+    if not hex_mac then
+      return nil, err
+    end
+    return hex_mac
+  end
+end
```
Comment (bungle) [inline lib/resty/session/utils.lua:827]: Almost no difference to (what is differences really to `hmac_sha256`):
https://github.com/bungle/lua-resty-session/blob/e7eab2060753bb01bb164c11db8fb1a7be457116/lib/resty/session/utils.lua#L769

**Review by bungle — CHANGES_REQUESTED** [review]

**Comment by bungle** [comment]
To be honest, I don't see any problem with this library. Should we have default pool name generation? Not sure. Should it be part of `resty.redis` library, why here?

**Comment by outsinre** [comment]
Deprecated by #179

**Pull request closed without merging**